contrib/src/test/java/com/datatorrent/contrib/parser/JsonParserTest.java

```diff
@@ -0,0 +1,356 @@
+        + "\"tags\": [\"cold\", \"ice\"]," + "\"dimensions\": {" + "\"length\": 7.0," + "\"width\" : 8.0,"
+        + "\"height\": 9.5" + "}," + "\"warehouseLocation\": {" + "\"latitude\": -78.75," + "\"longitude\": 20.4"
+        + "}," + "\"dateOfManufacture\": \"2013/09/29\"," + "\"dateOfExpiry\": \"2013\"" + "}";
+    parser.in.process(tuple);
+    Assert.assertEquals(1, objectPort.collectedTuples.size());
+    Assert.assertEquals(0, pojoPort.collectedTuples.size());
+    Assert.assertEquals(0, error.collectedTuples.size());
+    JSONObject jsonObject = (JSONObject)objectPort.collectedTuples.get(0);
+    Assert.assertEquals(2, jsonObject.getInt("id"));
+    Assert.assertEquals(1, jsonObject.getInt("price"));
+    Assert.assertEquals("An ice sculpture", jsonObject.get("name"));
+    Assert.assertEquals(7.0, jsonObject.getJSONObject("dimensions").getDouble("length"), 0);
+  }
+
+  @Test
+  public void testValidInputParsedOutputPortNotConnected() throws JSONException
+  {
+    parser.parsedOutput.setSink(null);
+    String tuple = "{" + "\"id\": 2," + "\"name\": \"An ice sculpture\"," + "\"price\": 1,"
+        + "\"tags\": [\"cold\", \"ice\"]," + "\"dimensions\": {" + "\"length\": 7.0," + "\"width\" : 8.0,"
+        + "\"height\": 9.5" + "}," + "\"warehouseLocation\": {" + "\"latitude\": -78.75," + "\"longitude\": 20.4"
+        + "}," + "\"dateOfManufacture\": \"2013/09/29\"," + "\"dateOfExpiry\": \"2013\"" + "}";
+    parser.in.process(tuple);
+    Assert.assertEquals(0, objectPort.collectedTuples.size());
+    Assert.assertEquals(1, pojoPort.collectedTuples.size());
+    Assert.assertEquals(0, error.collectedTuples.size());
+    Object obj = pojoPort.collectedTuples.get(0);
+    Assert.assertNotNull(obj);
+    Assert.assertEquals(Product.class, obj.getClass());
+    Product pojo = (Product)obj;
+    Assert.assertEquals(2, pojo.getId());
+    Assert.assertEquals(1, pojo.getPrice());
+    Assert.assertEquals("An ice sculpture", pojo.getName());
+    Assert.assertEquals(7.0, (double)pojo.getDimensions().get("length"), 0);
+  }
+
+  @Test
+  public void TestParserValidInputMetricVerification()
+  {
+    Assert.assertEquals(0, parser.parsedOutputCount);
+    String tuple = "{" + "\"id\": 2," + "\"name\": \"An ice sculpture\"," + "\"price\": 1,"
+        + "\"tags\": [\"cold\", \"ice\"]," + "\"dimensions\": {" + "\"length\": 7.0," + "\"width\" : 8.0,"
+        + "\"height\": 9.5" + "}," + "\"warehouseLocation\": {" + "\"latitude\": -78.75," + "\"longitude\": 20.4"
+        + "}," + "\"dateOfManufacture\": \"2013/09/29\"," + "\"dateOfExpiry\": \"2013\"" + "}";
+    parser.in.process(tuple);
+    Assert.assertEquals(1, parser.parsedOutputCount);
+  }
+
+  @Test
+  public void TestParserInvalidInputMetricVerification()
+  {
+    Assert.assertEquals(0, parser.parsedOutputCount);
+    String tuple = "{" + "\"id\": 2" + "}";
+    parser.in.process(tuple);
+    Assert.assertEquals(0, parser.parsedOutputCount);
+  }
```

Comment:
Sure @gauravgopi123 . Because beingWindow and endWindow did not have any functionality except resetting autometric which actually defaults to 0, i did not explicitly call them. But i agree with you, the test must reflect function calls in the same order as they are called in application. I have added it now.